### PR TITLE
Get the profile details to display from the meta data endpoint

### PR DESCRIPTION
**Pull request by thivi**

### CODE REVIEW

File: apps/user-portal/src/models/profile.ts [hunk 1]
```diff
@@ -16,23 +16,44 @@
  * under the License.
  */
 
+ /**
+  * Phone number model
```
Comment (jeradrutnam) [inline apps/user-portal/src/models/profile.ts]: Change it to `Phone numbers model`

File: apps/user-portal/src/store/actions/authenticate.ts [hunk 2]
```diff
@@ -75,10 +85,8 @@ export const getProfileInformation = () => {
                     dispatch(
                         fireNotification({
                             description:
-                                "views:components.profile.notifications.getProfileInfo.genericError" +
-                                ".description",
-                            message: "views:components.profile.notifications.getProfileInfo.genericError" +
-                                ".message",
+                                "views:components.profile.notifications.getProfileInfo.genericError" + ".description",
+                            message: "views:components.profile.notifications.getProfileInfo.genericError" + ".message",
```
Comment (jeradrutnam) [inline apps/user-portal/src/store/actions/authenticate.ts]: Fix indentation

File: apps/user-portal/src/models/profile.ts [hunk 3]
```diff
@@ -16,23 +16,44 @@
...
+  */
+export interface PhoneNumbers {
```
Comment (jeradrutnam) [inline apps/user-portal/src/models/profile.ts]: Make it `PhoneNumber`

File: apps/user-portal/src/models/profile.ts [hunk 4]
```diff
@@ -16,23 +16,44 @@
...
+    type: string;
+    value: number;
+}
+
+/**
+ * Roles model
+ */
+export interface Roles {
```
Comment (jeradrutnam) [inline apps/user-portal/src/models/profile.ts]: Change it to `Role`

File: apps/user-portal/src/models/profile.ts [hunk 5]
```diff
@@ -16,23 +16,44 @@
  * under the License.
  */
 
+ /**
+  * Phone number model
+  */
+export interface PhoneNumbers {
+    type: string;
+    value: number;
```
Comment (jeradrutnam) [inline apps/user-portal/src/models/profile.ts]: Change `value` => `number`

File: apps/user-portal/src/store/actions/authenticate.ts [hunk 6]
```diff
@@ -177,14 +187,27 @@ export const handleSignIn = () => {
...
-        SignOutUtil.sendSignOutRequest(LOGIN_CALLBACK_URL)
+        SignOutUtil.sendSignOutRequest(GlobalConfig.loginCallbackUrl)
             .then(() => {
                 dispatch(setSignOut());
                 AuthenticateSessionUtil.endAuthenticatedSession();
                 OPConfigurationUtil.resetOPConfiguration();
             })
             .catch
-            // TODO show error page.
+            // TODO: show error page.
```
Comment (jeradrutnam) [inline apps/user-portal/src/store/actions/authenticate.ts]: Fix indentation

File: modules/forms/package.json [hunk 7]
```diff
@@ -31,11 +31,11 @@
             "^.+\\.jsx$": "babel-jest"
         },
         "moduleNameMapper": {
-            "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/test_configs/file_mock.js",
-            "\\.(css|less)$": "<rootDir>/test_configs/style_mock.js"
+            "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/test-configs/file-mock.js",
```
Comment (jeradrutnam) [inline modules/forms/package.json]: Fix character length

File: modules/validation/src/validation.ts [hunk 8]
```diff
@@ -20,16 +20,41 @@
 import "fastestsmallesttextencoderdecoder";
 // tslint:disable-next-line:ordered-imports
 import Joi from "@hapi/joi";
+import { bool, string } from "prop-types";
 
-export const email = (value: string): boolean => {
+type ValidationFunction = (value: string) => boolean;
+/**
```
Comment (jeradrutnam) [inline modules/validation/src/validation.ts]: Keep a new line

File: apps/user-portal/src/api/profile.ts [hunk 9]
```diff
@@ -84,22 +84,25 @@ export const getProfileInfo = (): Promise<BasicProfileInterface> => {
...
                 organisation: response.data[orgKey] ? response.data[orgKey].organization : "",
                 phoneNumbers: response.data.phoneNumbers || [],
-                proUrl: response.data.profileUrl || "",
+                profileUrl: response.data.profileUrl || "",
                 responseStatus: response.status || null,
                 roles: response.data.roles || [],
-                userimage: response.data.userImage || gravatar,
-                username: response.data.userName || ""
+                userName: response.data.userName || "",
+                userimage: response.data.userImage || gravatar
```
Comment (inthirakumaaran) [inline apps/user-portal/src/api/profile.ts]:
```suggestion
                userImage: response.data.userImage || gravatar
```

Comment (jeradrutnam) [reply apps/user-portal/src/api/profile.ts]: @thivi - Please update the PR

Comment (jeradrutnam) [reply apps/user-portal/src/api/profile.ts]: Merging the PR, let's fix in the next PR

Comment (darshanasbg) [reply apps/user-portal/src/api/profile.ts]: We could have add the code suggestion, without need of the PR author right?

**Pull request merged**